Client/Frontend/Browser/ClipboardBarDisplayHandler.swift

```diff
@@ -0,0 +1,82 @@
+
+import Foundation
+import Shared
+
+public struct UserDefaultClipboardKey {
+    public static let KeyLastSavedURL = "KeylastSavedURL"
+}
+
+public struct ClipboardBarToastUX {
+    static let DismissAfter = 10.0
+}
+
+protocol ClipboardBarDisplayHandlerDelegate: class {
+    func shouldDisplayClipboardBar(absoluteString: String)
+}
+
+class ClipboardBarDisplayHandler {
+    weak var delegate: ClipboardBarDisplayHandlerDelegate?
+    private var sessionStarted = true
+    private var prefs: Prefs
+    private var lastDisplayedURL: String? {
+        if let value = UserDefaults.standard.object(forKey: UserDefaultClipboardKey.KeyLastSavedURL) as? String {
```

Comment:
I would prefer in memory just so it is simpler. It isn't a huge security concern because the URL isn't actually exposed to the user unless its already in their clipboard.